force-app/main/shared/classes/EventHandler.cls

```diff
@@ -0,0 +1,148 @@
+global class EventHandler extends MyTriggers {
+
+	global override void onAfterInsert() {
+		createEventShadow( (List<Event>) records );
+    }
+    
+    global override void onAfterUpdate( Map<Id, SObject> triggerOldMap ) {
+        updateEventShadow( (List<Event>) records);
+    }
+
+    global override void onAfterDelete() {
+        deleteEventShadow( (List<Event>) records );
+    }
+
+    global override void onAfterUndelete() {
+        undeleteEventShadow( (List<Event>) records );
+    }
+
+    private void createEventShadow( List<Event> events ) {
+
+        for(Event e : events ) {
+        
+        EventShadow__c es = new EventShadow__c();
+        es.SH_AccountOrgNumber__c=e.CRM_AccountOrgNumber__c;
+        es.SH_AccountParentId__c=e.TAG_AccountParentId__c;
+        es.SH_AccountParentOrgNumber__c=e.TAG_AccountParentOrgNumber__c;
+        es.SH_ActivityDateTime__c=e.ActivityDateTime;
+        es.SH_ActivityType__c=e.TAG_ActivityType__c;
+        es.SH_CreatedById__c=e.CreatedById;
+        es.SH_Date__c=e.ActivityDate;
+        es.SH_DateActivity__c=e.Date__c;
+        es.SH_DurationTime__c=e.DurationInMinutes;
+        es.SH_EndDateTime__c=e.EndDateTime;
+        es.SH_EventSubtype__c=e.EventSubtype;
+        es.SH_LastModifiedById__c=e.LastModifiedById;
+        es.SH_NoPersonalInformation__c=e.TAG_NoPersonInformation__c;
+        es.SH_OwnerId__c=e.OwnerId;
+        es.SH_Region__c=e.CRM_Region__c;
+        es.SH_RelatedToAccount__c=e.RelatedToAccount__c;
+        es.SH_Service__c=e.TAG_service__c;
+        es.SH_StartDateTime__c=e.StartDateTime;
+        es.SH_Status__c=e.TAG_Status__c;
+        es.SH_Subject__c=e.Subject;
+        es.SH_TaskEvent__c=e.TaskEvent__c;
+        es.SH_Type__c=e.Type;
+        es.SH_Unit__c=e.CRM_Unit__c;
+        es.SH_WhatId__c=e.WhatId;
+        es.SH_WhoId__c=e.WhoId;
+        es.SH_EventId__c=e.Id;
+
+        Insert es;
+        }
```

Comment:
```suggestion

    List<EventShadow__c> shadowEvents = new List<EventShadow__c>();

        for(Event e : events ) {
        
        EventShadow__c es = new EventShadow__c();
        es.SH_AccountOrgNumber__c=e.CRM_AccountOrgNumber__c;
        es.SH_AccountParentId__c=e.TAG_AccountParentId__c;
        es.SH_AccountParentOrgNumber__c=e.TAG_AccountParentOrgNumber__c;
        es.SH_ActivityDateTime__c=e.ActivityDateTime;
        es.SH_ActivityType__c=e.TAG_ActivityType__c;
        es.SH_CreatedById__c=e.CreatedById;
        es.SH_Date__c=e.ActivityDate;
        es.SH_DateActivity__c=e.Date__c;
        es.SH_DurationTime__c=e.DurationInMinutes;
        es.SH_EndDateTime__c=e.EndDateTime;
        es.SH_EventSubtype__c=e.EventSubtype;
        es.SH_LastModifiedById__c=e.LastModifiedById;
        es.SH_NoPersonalInformation__c=e.TAG_NoPersonInformation__c;
        es.SH_OwnerId__c=e.OwnerId;
        es.SH_Region__c=e.CRM_Region__c;
        es.SH_RelatedToAccount__c=e.RelatedToAccount__c;
        es.SH_Service__c=e.TAG_service__c;
        es.SH_StartDateTime__c=e.StartDateTime;
        es.SH_Status__c=e.TAG_Status__c;
        es.SH_Subject__c=e.Subject;
        es.SH_TaskEvent__c=e.TaskEvent__c;
        es.SH_Type__c=e.Type;
        es.SH_Unit__c=e.CRM_Unit__c;
        es.SH_WhatId__c=e.WhatId;
        es.SH_WhoId__c=e.WhoId;
        es.SH_EventId__c=e.Id;

        shadowEvents.add(es);
        }
        
        Insert shadowEvents;
```

**FY! Jeg vil aldri se en insert/update/delete/undelete inni for-loop! (det er helt normal feil å gjøre for nybegynnere, men den må du aldri glemme!)**

Legg det til i en liste, og insert hele listen istedenfor.